src/Connectors/src/ConnectorBase/Queue/RabbitMQHealthContributor.cs

```diff
@@ -49,19 +50,31 @@ public HealthCheckResult Health()
             var result = new HealthCheckResult();
             try
             {
-                var connection = ConnectorHelpers.Invoke(RabbitMQTypeLocator.CreateConnectionMethod, _connFactory, null);
+                connection ??= ConnectorHelpers.Invoke(RabbitMQTypeLocator.CreateConnectionMethod, _connFactory, null);
```

Comment:
I don't think this is right. You're injecting RabbitMQProviderConnectorFactory, which serves up instances of `ConnectionFactory`, which in turn are used to create an actual connection. It is the actual connection that you should be reusing, hence this should be a field not a local variable. Alternatively, the connection must be disposed of or it will keep TCP session open to the server till it's garbage callected and Dispose is called implicitly.

Reply:
Connection is stored in a field and is re-used... will look at enforcing `_` on fields in the future so it is more clear